### ISSUE TITLE
Treesitter folding not behaving correctly

### ISSUE THREAD

**Issue by RoRep1ay** [nvim-treesitter/nvim-treesitter#4152]
### Describe the bug

When running `zc` on a block after opening a new file, it close fold everything within the current file.

For example, a sample TS file containing:
```
function a() {
	return 'abc';
}

function b() {
	return 'hello';
}
```

Having the cursor on `a()` and running `zc` will close fold both functions. However, after this the folding behavior is working correctly.

### To Reproduce

1. Create a sample TS file with a few simple function in it
2. Open the newly created TS file
3. Run `zc` -> will now fold all the functions 🆖 
4. Run `zo` or `zc` -> will now behave properly 🆗 

### Expected behavior

After open the new TS file, running `zc` should *ONLY* close fold under the cursor not the whole file itself.

### Output of `:checkhealth nvim-treesitter`

```markdown
- INFO: Runtime ABI version : 14
  - OK: Loaded parser for bash: ABI version 14
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for dockerfile: ABI version 14
  - OK: Loaded parser for hcl: ABI version 14
  - OK: Loaded parser for help: ABI version 14
  - OK: Loaded parser for html: ABI version 13
  - OK: Loaded parser for javascript: ABI version 14
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for python: ABI version 14
  - OK: Loaded parser for typescript: ABI version 14
  - OK: Loaded parser for yaml: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for help: ABI version 14
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for vim: ABI version 14
```

### Output of `nvim --version`

```text
NVIM v0.8.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Ventura-arm64.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.8.2/share/nvim"

Run :checkhealth for more info
```

### Additional context

- I've test the folding on python languages and the behaviour seems to be the same.
- My Fold setting
  ```
  vim.o.foldmethod = "expr"
  vim.o.foldexpr = "nvim_treesitter#foldexpr()"
  vim.o.foldenable = false -- Disable folding at startup
  ```

**Comment by HerCerM** [comment]
This is (1) not a behavior specific to `nvim_treesitter#foldexpr()` and is (2) not a bug.

TL;DR: Use `vim.o.foldlevelstart = 99` instead of `vim.o.foldenable = false`.

### (1) Not a behavior specific to `nvim_treesitter#foldexpr()`

This behavior can be reproduced with a different fold method, e.g. `marker`.

1. Save the following to a file (here I leverage a `modeline` to open folds as you did):

```text
 vim:foldmethod=marker:nofoldenable

function a(){{{
  lines}}}

function b(){{{
  lines}}}
```

2. Open the file with the command: `nvim --clean {filename}`
3. Go to the line containing `function a(){{{`, type `zc`. Alternatively, type `zi`.

Actual behavior: All folds are closed.

### (2) Not a bug

The help for the option `'foldenable'` can be confusing as it says "When off, all folds are open". From a practical standpoint what it does is that it allows to enable/disable folds as a feature. This allows the workflow described in `:h usr_28.txt` (notice that `zi` just toggles the value of `'foldenable'`):

```text
You can quickly disable the folding with the |zn| command.  Then |zN| brings
back the folding as it was.  |zi| toggles between the two.  This is a useful
way of working:
- create folds to get overview on your file
- move around to where you want to do your work
- do |zi| to look at the text and edit it
- do |zi| again to go back to moving around
```

When a file is loaded and folds are enabled (`foldenable`), what controls which folds are opened or closed is the value of `'foldlevel'`, which by default is 0 which closes all folds. This is why in the reproduction steps above all folds can be closed by pressing `zi` so that the effect of `foldlevel=0` is seen. The folds were already closed.

To get the behavior that you expect, an approach is to set in your `init.lua` the following fold options. The last line sets `'foldlevel'` to 99, which opens all folds, whenever "starting to edit another buffer in a window":

```lua
vim.o.foldmethod = "expr"
vim.o.foldexpr = "nvim_treesitter#foldexpr()"
vim.o.foldlevelstart = 99
```

**Comment by RoRep1ay** [comment]
Hi. Thank you for your explaination. This is exactly what I'm looking for.

 Thanks a lot!

**Comment by HerCerM** [comment]
You're welcome!
